### PR TITLE
add eclipse-temurin and deprecate adoptopenjdk

### PR DESCRIPTION
**Pull request by gdams**
see https://github.com/docker-library/official-images/pull/10662 for context

### CODE REVIEW

File: adoptopenjdk/deprecated.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This image is officially deprecated in favor of [the `temurin-compliance` image](https://hub.docker.com/_/temurin-compliance/), and will receive no further updates after 2021-08-01 (Aug 01, 2021). Please adjust your usage accordingly.
```
Comment (tianon) [inline adoptopenjdk/deprecated.md:1]: `temurin-compliance` ? :eyes: :thinking:

Comment (tianon) [reply adoptopenjdk/deprecated.md:1]: It also seems kind of strange to actually deprecate this older image before the new one is actually a full, complete replacement?

Comment (gdams) [reply adoptopenjdk/deprecated.md]: fixed the name, thanks!

> It also seems kind of strange to actually deprecate this older image before the new one is actually a full, complete replacement?

I think deprecating now is the right thing to do, this repo will no longer be updated, the OpenJDK versions in that repo are now out of date and not recommended in production use. We will be adding the other architectures in the coming weeks so there won't be a big overlap

**Review by tianon — APPROVED** [review]

**Comment by tianon** [comment]
FYI affected image maintainers:

- `cassandra`: @tianon @yosifkit
- `gradle`: @keeganwitt
- `groovy`: @keeganwitt
- ~~`maven`: @carlossg~~ (https://github.com/docker-library/official-images/pull/10728)
- `open-liberty`: @crpotter @wraschke @arthurdm
- `tomcat`: @tianon @yosifkit
- `websphere-liberty`: @crpotter @wraschke @arthurdm

(This also helped catch https://github.com/docker-library/bashbrew/pull/32 :sweat_smile:)

**Pull request merged**